code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -18,65 +19,155 @@
 	return attack_hand(user, modifiers)
 
 /obj/structure/bed/wrench_act_secondary(mob/living/user, obj/item/weapon)
-	if(flags_1&NODECONSTRUCT_1)
+	if(flags_1 & NODECONSTRUCT_1)
 		return TRUE
+
 	..()
 	weapon.play_tool_sound(src)
 	deconstruct(disassembled = TRUE)
 	return TRUE
 
-/*
- * Roller beds
- */
-/obj/structure/bed/roller
-	name = "roller bed"
-	icon = 'icons/obj/medical/rollerbed.dmi'
-	icon_state = "down"
+/obj/structure/bed/post_buckle_mob(mob/living/buckled_mob)
+	occupied = TRUE
+
+/obj/structure/bed/post_unbuckle_mob(mob/living/unbuckled_mob)
+	occupied = FALSE
+
+/// Medical beds
+
+/obj/structure/bed/medical
+	name = "medical bed"
+	icon = 'icons/obj/medical/medical_bed.dmi'
+	desc = "A medical bed with wheels for assisted patient movement or medbay racing tournaments."
+	icon_state = "med_down"
+	base_icon_state = "med"
 	anchored = FALSE
 	resistance_flags = NONE
-	///The item it spawns when it's folded up.
-	var/foldabletype = /obj/item/roller
+	build_stack_type = /obj/item/stack/sheet/mineral/titanium
+	build_stack_amount = 1
+	/// The item it spawns when it's folded up.
+	var/foldable_type
+
+/obj/structure/bed/medical/anchored
+	anchored = TRUE
+
+/obj/structure/bed/medical/emergency
+	name = "emergency medical bed"
+	desc = "A compact medical bed. This emergency version can be folded and carried for quick transport."
+	icon_state = "emerg_down"
+	base_icon_state = "emerg"
+	foldable_type = /obj/item/emergency_bed
 
-/obj/structure/bed/roller/Initialize(mapload)
+/obj/structure/bed/medical/Initialize(mapload)
 	. = ..()
-	AddElement( \
-		/datum/element/contextual_screentip_bare_hands, \
-		rmb_text = "Fold up", \
-	)
 	AddElement(/datum/element/noisy_movement)
 
-/obj/structure/bed/roller/examine(mob/user)
+/obj/structure/bed/medical/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	if(held_item)
+		if(held_item.tool_behaviour != TOOL_WRENCH || flags_1 & NODECONSTRUCT_1)
+			return
+
+		context[SCREENTIP_CONTEXT_RMB] = "Dismantle"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	context[SCREENTIP_CONTEXT_ALT_LMB] = "[anchored ? "Release brakes" : "Apply brakes"]"
+	if(occupied)
+		context[SCREENTIP_CONTEXT_LMB] = "Unbuckle"
+
+	else if(!isnull(foldable_type))
+		context[SCREENTIP_CONTEXT_RMB] = "Fold up"
+
+	return CONTEXTUAL_SCREENTIP_SET
+
+/obj/structure/bed/medical/examine(mob/user)
+	. = ..()
+	if(anchored)
+		. += span_notice("The brakes are applied. They can be released with an Alt-click.")
+	else
+		. += span_notice("The brakes can be applied with an Alt-click.")
+
+	if(!isnull(foldable_type))
+		. += span_notice("You can fold it up with a Right-click.")
+
+/obj/structure/bed/medical/AltClick(mob/user)
+	. = ..()
+	anchored = !anchored
+	balloon_alert(user, "brakes [anchored ? "applied" : "released"]")
+	update_appearance()
+
+/obj/structure/bed/medical/post_buckle_mob(mob/living/patient)
+	set_density(TRUE)
+	occupied = TRUE
+	icon_state = "[base_icon_state]_up"
```

Comment:
If you want to change the icon_state (and in this case the pixel_y), do it in update_icon_state() and check for `buckled`
(this is already called when you call update_appearance)